adapter/slack.py

```diff
@@ -177,5 +183,38 @@ def bot_replied(self, event, parent_message):
         
         if chatbot is None:
             return {"text": self.MISSING_CHATBOT_ERROR}
-                
-        return asyncio.run(self.process_chatbot_request(chatbot, question, channel_id, thread_ts))
+        
+        history = self.get_chat_history(event)
+        return asyncio.run(self.process_chatbot_request(chatbot, question, channel_id, thread_ts, history))
+    
+    def get_chat_history(self, event):
+        all_messages = self.app.client.conversations_replies(
+            channel=event['channel'],
+            inclusive=True,
+            ts=event['thread_ts'],
+            include_all_metadata=True
+        )["messages"]
+        
+        result = []
+        INTRODUCTION_KEY_WORD = "asked:"
+
+        for i, message in enumerate(all_messages):
+            if i == 0:
+                question = self.extract_question(message, INTRODUCTION_KEY_WORD)
+                if question:
+                    result.append({"role": "user", "content": question})
+            else:
+                self.add_message_to_result(message, result)
+        
+        return result
+
+    def extract_question(self, message, key_word):
+        if 'text' in message and key_word in message['text']:
+            question_start = message['text'].find(key_word) + len(key_word) + 1
+            return message['text'][question_start:].strip().strip('"')
+        return None

```

Comment:
Will returning None cause problems down the line in the flow? Or is this expected behavior?

Reply:
Yeah, I think that could cause a problem.